src/Diagnostics.DataProviders/DataProviderContext.cs

```diff
@@ -11,7 +11,16 @@ public class DataProviderContext
         public string RequestId { get; private set; }
         public CancellationToken DataSourcesCancellationToken { get; private set; }
 
-        public DataProviderContext(DataSourcesConfiguration dataSourceConfiguration, string requestId = null, CancellationToken dataSourceCancellationToken = default(CancellationToken))
+        /// <summary>
+        /// Value of x-ms-client-object-id header received for requests coming from 'Diagnose and Solve' and Applens AppId for requests from Applens.
+        /// </summary>
+        public string clientObjectId {get; private set; }
```

Comment:
Because this is specific to the ChangeAnalysisDataProvider, I would suggest putting these properties into `ChangeAnalysisDataProviderConfiguration`

Reply:
`ChangeAnalysisDataProviderConfiguration` has configs for entire environment, right? These are not configs. These are detector specific. The values depend on where the detector is called from Applens vs Diagnose and Solve. I saw that `DataProviderContext` gets the value based on HttpContext. So I decided to keep them here